yb-voyager/src/srcdb/postgres.go

```diff
@@ -118,7 +119,7 @@ func (pg *PostgreSQL) getConnectionUri() string {
 		return source.Uri
 	}
 
-	source.Uri = fmt.Sprintf("postgresql://%s:%s@%s:%d/%s?%s", source.User, source.Password,
+	source.Uri = fmt.Sprintf("postgresql://%s:%s@%s:%d/%s?%s", url.QueryEscape(source.User), url.QueryEscape(source.Password),
 		source.Host, source.Port, source.DBName, generateSSLQueryStringIfNotExists(source))
```

Comment:
Now that I see it, `DBName` will have the same problem, and maybe `Host` too depending on how you process Internationalized Domain Names.
Additionally an ipv6 address passed as host will not work, you need `[::1]:5433` instead of `::1:5433`